include/SFML/Window/Cursor.hpp

```diff
@@ -0,0 +1,214 @@
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_CURSOR_HPP
+#define SFML_CURSOR_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Export.hpp>
+#include <SFML/System/NonCopyable.hpp>
+#include <SFML/System/Vector2.hpp>
+
+namespace sf
+{
+namespace priv
+{
+    class CursorImpl;
+}
+
+////////////////////////////////////////////////////////////
+/// \brief Cursor defines the appearance of a system cursor
+///
+////////////////////////////////////////////////////////////
+class SFML_WINDOW_API Cursor : NonCopyable
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Enumeration of the native system cursor types
+    ///
+    /// Refer to the following table to determine which cursor
+    /// is available on which platform.
+    ///
+    ///  Type                               | Linux | Mac OS X | Windows
+    /// ------------------------------------|:-----:|:--------:|:--------:
+    ///  sf::Cursor::Arrow                  |  yes  |    yes   |   yes
+    ///  sf::Cursor::ArrowWait              |  no   |    no    |   no
+    ///  sf::Cursor::Wait                   |  no   |    no    |   no
+    ///  sf::Cursor::Text                   |  no   |    yes   |   no
+    ///  sf::Cursor::Hand                   |  no   |    yes   |   no
+    ///  sf::Cursor::SizeHorizontal         |  no   |    yes   |   no
+    ///  sf::Cursor::SizeVertical           |  no   |    yes   |   no
+    ///  sf::Cursor::SizeTopLeftBottomRight |  no   |    no    |   no
+    ///  sf::Cursor::SizeBottomLeftTopRight |  no   |    no    |   no
+    ///  sf::Cursor::SizeAll                |  no   |    no    |   no
+    ///  sf::Cursor::Cross                  |  no   |    yes   |   no
+    ///  sf::Cursor::Help                   |  no   |    no    |   no
+    ///  sf::Cursor::NotAllowed             |  no   |    yes   |   no
+    ///
+    ////////////////////////////////////////////////////////////
```

Comment:
Ho, and the above table will need some update too. To the futur implementer: simply put a `yes` where appropriate once support for the given cursor is implemented on the given platform.